robot-agent/logMonitor.js

```diff
@@ -81,12 +81,51 @@ class LogMonitor {
         `${this.AGENT_PREFIX}/status/logs/lastLogTimestamp`
       ) || 0; // Get last log timestamp or default to 0
       this.initialized = true; // Set initialized state
-      log.info('Starting watching logs for packages registered before initialization');
-      _.forEach(this.watchedPackages, (packageData, packageName) => {
-        if( !packageData.initialized) {
-          this.watchLogs(packageName); // Start watching logs for the package
+      
+      // Subscribe to package topics to automatically start/stop monitoring
+      this.mqttSync.data.subscribePathFlat(`${this.AGENT_PREFIX}/status/package`, (value, topic, matched) => {
```

Comment:
It this really the right topic to watch? `runningPackages` may be saver, because it comes directly from the local mqtt brokers on-client-connect logic.

Reply:
The `startPackage` function (in main) handles this by first checking whether the package might already be running (by trying to find the PID). It only starts the package if it's not already running, i.e., when no such process can be found.